amulet/world_interface/__init__.py

```diff
@@ -66,22 +66,21 @@ def load_format(
 
     w = load_world(sys.argv[1])
     c = w.get_chunk(cx, cz)
-    for block in c.blocks.ravel()[:4096:16]:  # the blockstates of one vertical column
+    for block in c.blocks2[:4096:16].ravel():  # the blockstates of one vertical column
```

Comment:
I don't quite understand that code but what does that 4096 do here?

Reply:
You can ignore that. I have fixed it but haven't committed it yet. It should be `c.blocks2[0, :, 0].ravel()`

Reply:
What exactly is it doing? Is that a part that can be different when handling a specific world format? Whatever its doing won't work correctly with non standard height.

Reply:
That line in particular is just some debug code to test if things are working.
Lets use this `c.blocks2[0, :, 0].ravel()` because that is what I changed it to in a later commit and what is in this commit is invalid.
What it is doing is getting the column of blocks at x=0 and z=0 (in chunk coords) and the vertical height. Since there aren't values specified around the `:` for that axis it will default to `0:256` so what it actually does is get a vertical column of blocks from the chunk defaulting to the vanilla chunk size in the y direction.
If you wanted to specify a range outside of the vanilla area you can just put the numbers in like this `-10:300` and it should work.
The defaults are kind of there as a safeguard in case sub-chunk 0 is defined and a very large sub-chunk is defined since getattr returns a solid array and that array would be massive if `:` meant the real bounds of what is defined.